src/bsoncxx/lib/bsoncxx/v_noabi/bsoncxx/builder/core.cpp

```diff
@@ -686,6 +703,13 @@ core& core::close_array() {
     return *this;
 }
 
+core& core::close_binary() {
```

Comment:
`core->close_binary()` without a `core->open_binary()` is odd. Should one be added for symmetry?

```cpp
template <typename T>
detail::requires_t<void, detail::is_invocable<T, sub_binary>> generic_append(core* core, T&& func) {
    core->open_binary(); // <-- ?
    detail::invoke(std::forward<T>(func), sub_binary(core));
    core->close_binary();
}
```

Then again, support for mismatched open/close functions only seems strictly necessary to extend support to the stream builder, which I do not think we are interested in doing.